Fix FTBFS with gcc 4.9. #746857
--- a/src/jinglesession.cpp
+++ b/src/jinglesession.cpp
@@ -99,6 +99,10 @@
       }
     }
 
+    Session::Reason::~Reason()
+    {
+    }
+
     const std::string& Session::Reason::filterString() const
     {
       static const std::string filter = "jingle/reason";
--- a/src/jinglesession.h
+++ b/src/jinglesession.h
@@ -147,7 +147,7 @@
             /**
              * Virtual destructor.
              */
-            virtual ~Reason() {}
+            virtual ~Reason();
 
             /**
              * Returns the reason for the session termination.
